eppy/bunch_subclass.py

```diff
@@ -61,17 +63,39 @@ def extendlist(lst, i, value=''):
         lst.extend([value, ] * (i - len(lst) + 1))
 
 
-
 def return42(self, *args, **kwargs):
     # proof of concept - to be removed
     return 42
 
+
+def _register_function(name, cls, keys):
+    def decorator(function):
+        if hasattr(cls, name):
+            warnings.warn(
+                "registration of function {!r} under name {!r} for type "
+                "{!r} is overriding a preexisting attribute with the same "
+                "name.".format(function, name, cls),
+                UserWarning,
+                stacklevel=2,
+            )
+        setattr(cls, name, function)
+        cls._EpBunch__functions.update({name: (function, keys)})
+        return function
+
+    return decorator
+
+
+def register_epbunch_function(name, keys):
```

Comment:
Users call this method as a decorator `@` and specify the name of their function and the keys of the EP objects this method should work with.